sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/PersistentStorage/StorageHelper.cs

```diff
@@ -29,50 +25,18 @@ internal static string GetDefaultStorageDirectory()
 
                 if (IsWindowsOS())
                 {
-                    string localAppData = environmentVars["LOCALAPPDATA"]?.ToString();
-                    if (localAppData != null)
-                    {
-                        dirPath = CreateTelemetryDirectory(localAppData);
-                        if (dirPath != null)
-                        {
-                            s_defaultStorageDirectory = dirPath;
-                            return s_defaultStorageDirectory;
-                        }
-                    }
-
-                    string temp = environmentVars["TEMP"]?.ToString();
-                    if (temp != null)
+                    if (TryCreateTelemetryDirectory(path: environmentVars["LOCALAPPDATA"]?.ToString(), createdDirectoryPath: out dirPath)
```

Comment:
This or statement makes it very difficult read the code. Considering the removal of nested statement we are fine.

Reply:
discussed offline and decided to leave as is.

Changing this to a loop would be the cleanest implementation but would have some allocations that we're deciding to avoid here.